core/validation/specific_validations/towns_fund_round_four.py

```diff
@@ -448,6 +450,42 @@ def get_allocated_funding():
     )["Grant Awarded"]
 
 
+def validate_psi_funding_not_negative(
+    workbook: dict[str, pd.DataFrame]
+) -> list["TownsFundRoundFourValidationFailure"] | None:
+    """
+    Validates that Private Sector Funding amounts are not negative.
+
+    This function checks the 'Private Investments' sheet for rows where either
+    'Private Sector Funding Required' or 'Private Sector Funding Secured' is negative.
+    If invalid rows are found, it returns a list of validation failures.
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas
+                     DataFrames representing each sheet in the Round 4 submission.
+    :return: ValidationErrors or None if no errors are found.
+    """
+    psi_df = workbook["Private Investments"]
+    cols_to_check = ("Private Sector Funding Required", "Private Sector Funding Secured")
+    errors = []
+    for col in cols_to_check:
+        # coerce to numeric so that value can be checked if less than 0, and error not raised on strings
+        for index, val in pd.to_numeric(psi_df[col], errors="coerce").iteritems():
+            if val < 0:
+                errors.append((col, index))
```

Comment:
looks like this could be a list comprehension - perhaps it wouldn't be very readable - have you tried it?

Reply:
```
errors = [
        (col, index)
        for col in cols_to_check
        for index, val in pd.to_numeric(psi_df[col], errors="coerce").iteritems()
        if val < 0
    ]
```
Readable enough. Amended and pushed

Reply:
nice